src/order/ideal.lean

```diff
@@ -93,6 +106,10 @@ instance : has_coe (ideal P) (set P) := ⟨carrier⟩
 /-- For the notation `x ∈ I`. -/
 instance : has_mem P (ideal P) := ⟨λ x I, x ∈ (I : set P)⟩
 
+@[simp] lemma mem_def : x ∈ I ↔ x ∈ (I : set P) := iff_of_eq rfl
```

Comment:
Do we really want this to be a simp lemma? I think, for instance, that it will prevent `mem_principal` from ever triggering under simp.

Reply:
I'm not entirely sure, maybe you're right

Reply:
Either that, or you could tweak the statement of `mem_principal`.